x/gov/client/rest/util.go

```diff
@@ -0,0 +1,199 @@
+package rest
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	"github.com/pkg/errors"
+)
+
+//type request interface {
+//	Validate(w http.ResponseWriter) bool
+//}
+
+type baseReq struct {
```

Comment:
commonReq?

Reply:
What's wrong with baseReq?

Reply:
Also, we should probably move this baseReq thing to auth module?

Reply:
totally agree on moving to the auth module. - yeah `baseReq` seems fine was just an idea